model.py

```diff
@@ -1,103 +1,1222 @@
+
+        # GP hyperparameters
+        self.ampin = ampin_GP
+        self.ampout = ampout_GP
+        self.tau = tau_GP
+
+        super().__init__(*args, **kwargs)
+
+    def calcChangepoints(self):
+        '''Caclulate the WD ingress and egresses, i.e. where we want to switch
+        on or off the extra GP amplitude'''
+
+        # Also get object for dphi, q and rwd as this is required to determine
+        # changepoints
+        dphi = self.par_dict['dphi']
+        q = self.par_dict['q']
+        rwd = self.par_dict['rwd']
+        phi0 = self.par_dict['phi0']
+
+        dphi_change = np.fabs(self._olddphi - dphi.currVal) / dphi.currVal
+        q_change = np.fabs(self._oldq - q.currVal) / q.currVal
+        rwd_change = np.fabs(self._oldrwd - rwd.currVal) / rwd.currVal
+
+        # Check to see if our model parameters have changed enough to
+        # significantly change the location of the changepoints.
+        if (dphi_change > 1.2) or (q_change > 1.2) or (rwd_change > 1.2):
+            # Calculate inclination
+            inc = roche.findi(q.currVal, dphi.currVal)
+
+            # Calculate wd contact phases 3 and 4
+            phi3, phi4 = roche.wdphases(q.currVal, inc, rwd.currVal, ntheta=10)
+
+            # Calculate length of wd egress
+            dpwd = phi4 - phi3
+
+            # Distance from changepoints to mideclipse
+            dist_cp = (dphi.currVal+dpwd)/2.
+
+            # save these values for speed
+            self._dist_cp = dist_cp
+            self._oldq = q.currVal
+            self._olddphi = dphi.currVal
+            self._oldrwd = rwd.currVal
+        else:
+            # Use the old values
+            dist_cp = self._dist_cp
+
+        # Find location of all changepoints
+        min_ecl = int(np.floor(self.lc.x.min()))
+        max_ecl = int(np.ceil(self.lc.x.max()))
+
+        eclipses = [e for e in range(min_ecl, max_ecl+1)
+                    if np.logical_and(e > self.lc.x.min(),
+                                      e < 1+self.lc.x.max()
+                                      )
+                    ]
+
+        changepoints = []
+        for e in eclipses:
+            # When did the last eclipse end?
+            egress = (e-1) + dist_cp
+            # When does this eclipse start?
+            ingress = e - dist_cp
+            changepoints.append([egress, ingress])
+
+        return changepoints
+
+    def createGP(self):
+        """Constructs a kernel, which is used to create Gaussian processes.
+
+        Using values for the two hyperparameters (amp,tau), amp_ratio and dphi,
+        this function: creates kernels for both inside and out of eclipse,
+        works out the location of any changepoints present, constructs a single
+         (mixed) kernel and uses this kernel to create GPs"""
+
+        # Get objects for ampin_gp, ampout_gp, tau_gp and find the exponential
+        # of their current values
+        ln_ampin = self.ampin
```

Comment:
careful about mixing logs and not-logs here. If ```self.ampin``` is meant to be the logarithm, give it an appropriate name

Reply:
The input files all refer to ampin, ampout and tau without the ln prefix, so I mirrored that in my version so that the input files need minimal tweaking. Thinking about it, it wouldn't be much to write a few lines to interpret old stuff properly, and new stuff can be ln_XXX. Something like;
```
input_dict = ...
for key, item in input_dict.items():
    if key in ['ampin_gp', 'ampout_gp', 'tau_gp']:
        print("Scold the user for their bad practice")
        input_dict['ln_'+key] = item
```

Then the rest of the code can refers to ln_XXX.